src/gateways/RedisDraftRegistrationGateway.ts

```diff
@@ -0,0 +1,46 @@
+import Redis from "ioredis";
+import JSONCache from "redis-json";
+import { DraftRegistration } from "../entities/DraftRegistration";
+import { DraftRegistrationGateway } from "./DraftRegistrationGateway";
+
+export class RedisDraftRegistrationGateway implements DraftRegistrationGateway {
+  private cache = new JSONCache<DraftRegistration>(
+    new Redis(process.env.REDIS_URI)
+  );
+
+  public async deleteUse(
+    submissionId: string,
+    useIndex: number
+  ): Promise<void> {
+    const registration: DraftRegistration = await this.read(submissionId);
+
+    const registrationMinusDeletedUse = {
+      ...registration,
+      uses: registration.uses.filter((use, i) => i !== useIndex),
+    };
+
+    await this.update(submissionId, registrationMinusDeletedUse);
+  }
+
+  public async read(id: string): Promise<DraftRegistration> {
+    return (await this.cache.get(id)) as Promise<DraftRegistration>;
+  }
+
+  public async update(
+    id: string,
+    draftRegistration: DraftRegistration
+  ): Promise<void> {
+    await this.cache.set(id, draftRegistration);
+  }
+
+  public async createEmptyUse(submissionId: string): Promise<void> {
+    const registration: DraftRegistration = await this.read(submissionId);
+
+    const registrationWithNewUse = {
+      ...registration,
+      uses: [...(registration?.uses || []), {}],
+    };
+
+    await this.update(submissionId, registrationWithNewUse);
+  }
+}
```

Comment:
Rename from `RedisCachedRegistrationGateway` to `RedisDraftRegistrationGateway` and remove uses of `FormCache` (which depended on the concrete `Registration` class we're moving away from) in favour of direct use of the Redis client.